framework/src/kernels/Kernel.C

```diff
@@ -80,6 +80,25 @@ Kernel::computeJacobian()
 
   ke += _local_ke;
 
+  if (_calculate_nonlocal_variable_jacobian)
+  {
+    _nonlocal_ke.resize(ke.m(), _subproblem.es().n_dofs());
+    _nonlocal_ke.zero();
+    for (_i = 0; _i < _test.size(); _i++)
```

Comment:
There's a way through the dof_map somewhere... I don't know for sure though. You might ask this question on the list. Very few people watch the tickets.